lib/matplotlib/ticker.py

```diff
@@ -1685,6 +1685,33 @@ def __init__(self, *args, **kwargs):
         self.set_params(**self.default_params)
         self.set_params(**kwargs)
 
+    @staticmethod
+    def _validate_steps(steps):
+        if not np.iterable(steps):
+            raise ValueError('steps argument must be a sequence of numbers '
+                             'from 1 to 10')
+        steps = np.asarray(steps)
+        if np.any(np.diff(steps) <= 0):
+            raise ValueError('steps argument must be uniformly increasing')
+        if np.any((steps > 10) | (steps < 1)):
```

Comment:
Since you've already checked that it's increasing, you only need to check the first/last elements, I think.